modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InputParametersTypeModelValidator.java

```diff
@@ -53,6 +55,8 @@
  */
 public final class InputParametersTypeModelValidator implements ExtensionModelValidator {
 
+  private ReflectionCache reflectionCache = new ReflectionCache();
```

Comment:
we avoid state in validators and loaders since they're never disposed. check this doesn't generate a new leak and if it cannot be parameterized instead